src/apply.jl

```diff
@@ -1,180 +1,92 @@
-    @eval begin
-        function ($op){N}(ta::TimeArray{Bool,N}, var::Bool)
-            cnames = [name * string($op) * string(var) for name in ta.colnames]
-            vals   = ($op)(ta.values, var)
-            TimeArray(ta.timestamp, vals, cnames, ta.meta)
-        end # function
-    end # eval
-end # loop
-
-# Bool <--> TimeArray
-for op in BOOLEAN_OPS
-    @eval begin
-        function ($op){N}(var::Bool, ta::TimeArray{Bool,N})
-            cnames = [string(var) * string($op) * name for name in ta.colnames]
-            vals   = ($op)(var, ta.values)
-            TimeArray(ta.timestamp, vals, cnames, ta.meta)
-        end # function
-    end # eval
-end # loop
-
-# Boolean ND TimeArray <--> Boolean ND TimeArray
-# Doesn't support broadcasting since it isn't supported by Base either
-for op in BOOLEAN_OPS
-    @eval begin
-        function ($op){N}(ta1::TimeArray{Bool,N}, ta2::TimeArray{Bool,N})
-            # test column count matches
-            length(ta1.colnames) == length(ta2.colnames) ?
-              ncols=length(ta1.colnames) :
-              error("arrays must have the same number of columns")
-            # test metadata matches
-            ta1.meta == ta2.meta ? meta = ta1.meta : error("metadata doesn't match")
-            cnames = [ta1.colnames[i]*string($op)*ta2.colnames[i] for i=1:ncols]
-            idx1, idx2 = overlaps(ta1.timestamp, ta2.timestamp)
-            # obtain shared timestamp
-            tstamp = ta1[idx1].timestamp
-            # retrieve values that match the Int array matching dates
-            vals1  = ta1[idx1].values
-            vals2  = ta2[idx2].values
-            vals = ($op)(vals1, vals2)
-            TimeArray(tstamp, vals, cnames, meta)
-        end # function
-    end # eval
-end # loop
+function broadcast(f, ta1::TimeArray, ta2::TimeArray)
+    # first test metadata matches
+    meta = ta1.meta == ta2.meta ? ta1.meta : Void
+
+    # determine array widths and name cols accordingly
+    w1, w2  = length(ta1.colnames), length(ta2.colnames)
+    if w1 == w2
+        cnames = [ta1.colnames[i] * "_" * ta2.colnames[i] for i = 1:w1]
+    elseif w1 == 1
+        cnames = [ta1.colnames[1] * "_" * ta2.colnames[i] for i = 1:w2]
+    elseif w2 == 1
+        cnames = [ta1.colnames[i] * "_" * ta2.colnames[1] for i = 1:w1]
+    else
+        error("arrays must have the same number of columns, or one must be a single column")
+    end
+
+    # obtain shared timestamp
+    idx1, idx2 = overlaps(ta1.timestamp, ta2.timestamp)
+    tstamp = ta1[idx1].timestamp
+
+    # retrieve values that match the Int array matching dates
+    vals1, vals2 = ta1[idx1].values, ta2[idx2].values
+
+    # compute output values
+    vals = broadcast(f, vals1, vals2)
+    TimeArray(tstamp, vals, cnames, meta)
+end
+
+function broadcast(f, ta::TimeArray, args...)
+    vals = broadcast(f, ta.values, args...)
+    TimeArray(ta.timestamp, vals, ta.colnames, ta.meta)
+end
+
+# FIXME: How to deal with f(Number, Number, ..., TimeArray)?
```

Comment:
I think we should  do the integration like this: https://github.com/JuliaArrays/StaticArrays.jl/pull/241

Maybe another PR to make it? seems quite complicated